src/Registry.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+error AlreadyAdded();
+
+/// @title  Registry contract :: Adding popular token's data :: DAO Owned 
+/// @notice This contract can be called by our main contract for fetching tokenData
+
+contract Registry {
+
+    ///  Address of the DAO
+    address immutable owner_DAO;
+
+    /// This struct contains necessary info about an ERC20 token
+    /// What more params could be added?
+    struct Token{
+        string name;
+        string symbol;
+        uint8 UID;
+        uint8 decimals;
+    }
+    
+    /// Registry mapping ERC20 token address to its metadata
+    mapping(address=>Token) tokenData;
```

Comment:
I don't think this is necessary!

All this information is already available on chain! Each ERC20 has all these methods, and we could call them from our rendering contract.